### PR TITLE
opencv3: fixes opencl build error

### PR DESCRIPTION
**Pull request by heinemml**
### Have you:

- [x] Followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-science/blob/master/.github/CONTRIBUTING.md) document?
- [x] Checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-science/pulls) for the same update/change?
- [x] Run `brew audit --strict --online <formula>` (where `<formula>` is the name of the formula you're submitting) and corrected all errors?
- [x] Built your formula locally prior to submission with `brew install <formula>`?

### Updates to existing formula: have you

- [x] Added/bumped the `revision` number if the changes affect the precompiled bottles?
- [x] Not altered the `bottle` section?
- [x] Checked that the tests still pass?

patch contains upstream commit https://github.com/Itseez/opencv/commit/c7bdbef5042dadfe032dfb5d80f9b90bec830371 which fixes https://github.com/Itseez/opencv/issues/6052
this fixes https://github.com/Homebrew/homebrew-science/issues/3147

### CODE REVIEW

File: opencv3.rb [hunk 1]
```diff
@@ -13,7 +13,9 @@ class Opencv3 < Formula
     end
     # patch fixing crash after 100s when using capturing device https://github.com/Itseez/opencv/issues/5874
     # fixed in https://github.com/Itseez/opencv/commit/a2bda999211e8be9fbc5d40038fdfc9399de31fc
-    # this can be removed when a new version is released
+    # 2nd patch fixes build error https://github.com/Homebrew/homebrew-science/issues/3147 when not using --without-opencl
+    # fixed in https://github.com/Itseez/opencv/commit/c7bdbef5042dadfe032dfb5d80f9b90bec830371
+    # both patches can be removed when a new version is released
```
Comment (dpo) [inline opencv3.rb]: Is there any chance of grabbing the patch from `https://github.com/Itseez/opencv/commit/c7bdbef5042dadfe032dfb5d80f9b90bec830371.diff`?

Comment (heinemml) [inline opencv3.rb]: Sure, didn't find a guideline on which way to prefer. I will change both patches to pull it from github.

**Comment by heinemml** [comment]
I changed the commit to pull all patches directly from upstream.
I also fixed the style warnings issued by 'brew audit'. Is this okay or would this be better in an additional commit and/or pull request?

**Comment by ahundt** [comment]
can this be merged soon?

**Comment by dpo** [comment]
@BrewTestBot  test this please

**Comment by heinemml** [comment]
From my side this PR is ready. The CI build was sucessful before but currently the brewbot seems to be in some kind of deadlock state by some changes from core...

**Comment by ahundt** [comment]
@dpo ready to merge perhaps? Thanks!

**Pull request closed without merging**